pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1706,6 +1706,64 @@ func PublishVolumeToZedCloud(ctx *zedagentContext, uuid string,
 	}
 }
 
+// PublishBlobInfoToZedCloud is called per change, hence needs to try over all management ports
+// When blob Status is nil it means a delete and we send a message
+// containing only the UUID to inform zedcloud about the delete.
+func PublishBlobInfoToZedCloud(ctx *zedagentContext, blobSha string, blobStatus *types.BlobStatus, iteration int) {
+	log.Infof("PublishBlobInfoToZedCloud blobSha %v", blobSha)
+	var ReportInfo = &info.ZInfoMsg{}
+
+	contentType := new(info.ZInfoTypes)
+	*contentType = info.ZInfoTypes_ZiBlobList
+	ReportInfo.Ztype = *contentType
+	ReportInfo.DevId = *proto.String(zcdevUUID.String())
+	ReportInfo.AtTimeStamp = ptypes.TimestampNow()
+
+	ReportBlobInfoList := new(info.ZInfoBlobList)
+
+	ReportBlobInfo := new(info.ZInfoBlob)
+
+	ReportBlobInfo.Sha256 = blobSha
+	if blobStatus != nil {
+		ReportBlobInfo.State = blobStatus.State.ZSwState()
+		ReportBlobInfo.ProgressPercentage = uint32(blobStatus.ProgressPercentage)
+		ReportBlobInfo.Usage = &info.UsageInfo{RefCount: uint32(blobStatus.RefCount)}
+	}
+	ReportBlobInfoList.Blob = append(ReportBlobInfoList.Blob, ReportBlobInfo)
+
+	ReportInfo.InfoContent = new(info.ZInfoMsg_Binfo)
+	if blobInfoList, ok := ReportInfo.GetInfoContent().(*info.ZInfoMsg_Binfo); ok {
+		blobInfoList.Binfo = ReportBlobInfoList
+	}
+
+	log.Infof("PublishBlobInfoToZedCloud sending %v", ReportInfo)
+
+	//data, err := proto.Marshal(ReportInfo)
+	//if err != nil {
+	//	log.Fatal("PublishBlobInfoToZedCloud proto marshaling error: ", err)
+	//}
+	//statusURL := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, devUUID, "info")
+	//
+	//zedcloud.RemoveDeferred(blobSha)
+	//buf := bytes.NewBuffer(data)
+	//if buf == nil {
+	//	log.Fatal("malloc error")
+	//}
+	//size := int64(proto.Size(ReportInfo))
+	//err = SendProtobuf(statusURL, buf, size, iteration)
+	//if err != nil {
+	//	log.Errorf("PublishBlobInfoToZedCloud failed: %s", err)
+	//	// Try sending later
+	//	// The buf might have been consumed
+	//	buf := bytes.NewBuffer(data)
+	//	if buf == nil {
+	//		log.Fatal("malloc error")
+	//	}
+	//	zedcloud.SetDeferred(blobSha, buf, size, statusURL, zedcloudCtx,
+	//		true)
+	//}
```

Comment:
Shouldn't we start sending this to the controller? Even if it can't handle deletes yet.